docs/framework/guides/tutorials/data-from-external-source.md

```diff
@@ -0,0 +1,886 @@
+
+		this.editor.commands.add( 'external', new ExternalDataWidgetCommand( this.editor ) );
+	}
+
+	// Interval function
+	_intervalFetch() {
+		return setInterval( () => this._updateWidgetData(), 15000 ); // set time interval to 15s
+	}
+
+	// Fetch data and update all widget instances
+	async _updateWidgetData( externalUrl = RESOURCE_URL ) {
+		try {
+			const response = await fetch( externalUrl );
+			const data = await response.json();
+			const updateTime = new Date( data.closeTime );
+
+			// Example parsed data: $17098.35 - 09/11/2022, 18:04:18
+			const parsedData = '$' + Number( data.lastPrice ).toFixed( 2 ) + ' - ' + updateTime.toLocaleString();
+
+			// Update property with last fetched and parsed data
+			this.externalDataValue = parsedData;
+
+			const rootElement = this.editor.model.document.getRoot();
+
+			// Iterate over whole editor content, search for external data widget instances
+			// and trigger `recovertItem` function
+			for ( const { item } of this.editor.model.createRangeIn( rootElement ) ) {
+				if ( item.is( 'element', 'externalElement' ) ) {
+					this.editor.editing.reconvertItem( item );
+				}
+			}
+		} catch ( error ) {
+			console.error( error );
+		}
+	}
+
+	_defineSchema() {
+		// ...
+	}
+
+	_defineConverters() {
+		// ...
+
+		editor.conversion.for( 'editingDowncast' ).elementToElement( {
+			model: 'externalElement',
+			view: ( modelElement, { writer } ) => {
+				const externalValueToShow = this.externalDataValue;
+
+				const externalDataPreviewElement = writer.createRawElement( 'span', null, function( domElement ) {
+					// When the value is not present (initial run) show a placeholder
+					domElement.textContent = externalValueToShow || 'Fetching data...';
+
+					// If a new value arrives, add a CSS animation effect to show that data were updated
+					if ( externalValueToShow ) {
+						domElement.classList.add( 'external-data-widget-bounce' );
+						// Remove the animation class when it ends
+						setTimeout( () => domElement.classList.remove( 'external-data-widget-bounce' ), 1100 );
+					}
+				} );
+
+				const externalWidgetContainer = writer.createContainerElement( 'span', null, externalDataPreviewElement );
+
+				return toWidget( externalWidgetContainer, writer, {
+					label: 'External widget'
+				} );
+			}
+		} );
+	}
+}
+
+```
+
+The editor content traversal can be a challenging process. The presented method is sufficient when there is relatively little content. Otherwise, a `WeakMap` will be a better option.
+
+
+## Demo
+
+You can see the external data widget implementation in action in the editor below.
+
+{@snippet framework/tutorials/external-data-widget}
+
+## Final solution
+
+The following code snippet contains the complete implementation of the `ExternalDataWidget` plugin (and all of its dependencies) and the code needed to run the editor. You can paste it into the `app.js` file and it will run out–of–the–box (excluded the Bitcoin logo):
+
+```js
```

Comment:
Removing an icon from this fully worked snippet is done on purpose (small mention about it is at the end: `(excluded the Bitcoin logo)`).  
Like it is written its a `copy/paste` code example. When import of the svg icon will be added - it will cause errors when user wont have that icon in his directory. (also the icon is removed from theButtonView - left `withText: true`)

Of course i can add it, but first can mention about it above the code snippet, so the user needs to be sure that the icon is there where the import is made from.

WDYT??